core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -979,16 +979,15 @@ protected void doRun() throws Exception {
                     for (ComputerListener cl : ComputerListener.all()) {
                         try {
                             cl.onOnline(c, new LogTaskListener(LOGGER, INFO));
-                        } catch (Throwable t) {
-                            if (t instanceof Error) {
-                                // We propagate Runtime errors, because they are fatal.
-                                throw t;
-                            }
-
-                            // Other exceptions should be logged instead of failing the Jenkins startup (See listener's Javadoc)
-                            // We also throw it for InterruptedException since it's what is expected according to the javadoc
-                            LOGGER.log(SEVERE, String.format("Invocation of the computer listener %s failed for the Jenkins master node",
-                                    cl.getClass()), t);
+                        } catch (IOException | InterruptedException e) {
+                            // Per Javadoc log exceptions but still go online.
+                            LOGGER.log(WARNING, String.format("Exception in onOnline() for the computer listener %s on the Jenkins master node",
+                                cl.getClass()), e);
+                        } catch (Exception e) {
```

Comment:
Looks like the only difference here is the log level. I'm not sure there's an easy way to combine these and keep the different log levels, but I'm not sure the different log levels are necessary.

Reply:
Good point.  The SEVERE preserve the existing behavior, but I can change them both to warning.